motor/discovery/gcp/mql_assets.go

```diff
@@ -0,0 +1,174 @@
+	}
+	if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, DiscoveryComputeFirewalls) {
+		assets = append(assets, computeFirewalls(m, project, tc)...)
+	}
+	if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, DiscoveryGkeClusters) {
+		assets = append(assets, gkeClusters(m, project, tc)...)
+	}
+	if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, DiscoveryStorageBuckets) {
+		assets = append(assets, storageBuckets(m, project, tc)...)
+	}
+	if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, DiscoveryBigQueryDatasets) {
+		assets = append(assets, bigQueryDatasets(m, project, tc)...)
+	}
+
+	return assets, nil
+}
+
+type mqlObject struct {
+	name      string
+	labels    map[string]string
+	gcpObject gcpObject
+}
+
+type gcpObject struct {
+	project    string
+	region     string
+	id         string
+	service    string
+	objectType string
+	name       string
+}
+
+type gcpObjectPlatformInfo struct {
+	title    string
+	platform string
+}
+
+func GcpPlatformID(o gcpObject) string {
+	return "//platformid.api.mondoo.app/runtime/gcp/" + o.service + "/v1/projects/" + o.project + "/regions/" + o.region + "/" + o.objectType + "/" + o.name
+}
+
+func MqlObjectToAsset(account string, mqlObject mqlObject, tc *providers.Config) *asset.Asset {
+	if mqlObject.name == "" {
+		mqlObject.name = mqlObject.gcpObject.id
+	}
+	if err := validate(mqlObject); err != nil {
+		log.Error().Err(err).Msg("missing values in mql object to asset translation")
+		return nil
+	}
+	info, err := getTitleFamily(mqlObject.gcpObject)
+	if err != nil {
+		log.Error().Err(err).Msg("missing runtime info")
+		return nil
+	}
+	platformid := GcpPlatformID(mqlObject.gcpObject)
+	t := tc.Clone()
+	t.PlatformId = platformid
+	return &asset.Asset{
+		PlatformIds: []string{platformid},
+		Name:        mqlObject.name,
+		Platform: &platform.Platform{
+			Name:    info.platform,
+			Title:   info.title,
+			Kind:    providers.Kind_KIND_GCP_OBJECT,
+			Runtime: providers.RUNTIME_GCP,
+		},
+		State:       asset.State_STATE_ONLINE,
+		Labels:      addInformationalLabels(mqlObject.labels, mqlObject),
+		Connections: []*providers.Config{t},
+	}
+}
+
+func validate(m mqlObject) error {
+	if m.name == "" {
+		return errors.New("name required for mql gcp object to asset translation")
+	}
+	if m.gcpObject.id == "" {
+		return errors.New("id required for mql gcp object to asset translation")
+	}
+	if m.gcpObject.region == "" {
+		return errors.New("region required for mql gcp object to asset translation")
+	}
+	if m.gcpObject.project == "" {
+		return errors.New("project required for mql gcp object to asset translation")
+	}
+	if m.gcpObject.name == "" {
+		return errors.New("name required for mql gcp object to asset translation")
+	}
+	return nil
+}
+
+func addInformationalLabels(l map[string]string, o mqlObject) map[string]string {
+	if l == nil {
+		l = make(map[string]string)
+	}
+	l[RegionLabel] = o.gcpObject.region
+	l[common.ParentId] = o.gcpObject.project
```

Comment:
Is this for the relations?

I have the example of project -> network -> sub-network in my mind. There the direct parent would not be the project but the network.

Reply:
I took that from #804. @vjeffrey what was your idea with that label?

Reply:
i use that for general grouping -- e.g. if you scan two diff aws accounts, its hard to tell what's what in the ui. this allows us to filter by a single account - so its not so much about relationships and more about grouping

Reply:
sounds like the parent is then just a project (GCP)/account (AWS)